mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessage.java

```diff
@@ -68,12 +68,12 @@ public static IndexableMessage from(MailboxMessage message, List<User> users, Te
     }
 
     private void setFlattenedAttachments(MimePart parsingResult, IndexAttachments indexAttachments) {
-        if (indexAttachments.equals(IndexAttachments.YES)) {
-            attachments = parsingResult.getAttachmentsStream()
-                    .collect(Collectors.toList());
-        } else {
-            attachments = ImmutableList.of();
-        }
+        List<MimePart> mimeparts = parsingResult.getAttachmentsStream()
+                .collect(Collectors.toList());
+
+        hasAttachment = mimeparts.stream().anyMatch(IndexableMessage::isAttachmentMimepart);
```

Comment:
hasAttachment is a "view" of attachments so it's not a problem to set both here.
The question here : do you really need hasAttachment to be a field ? I think that a method would do.